src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Builder/OpenTelemetryBuilderOtlpExporterExtensions.cs

```diff
@@ -0,0 +1,210 @@
+        OtlpExportProtocol protocol,
+        Uri baseEndpoint)
+    {
+        Guard.ThrowIfNull(baseEndpoint);
+
+        return UseOtlpExporter(builder, name: null, configure: otlpBuilder =>
+        {
+            otlpBuilder.ConfigureDefaultExporterOptions(o =>
+            {
+                o.Protocol = protocol;
+                o.Endpoint = baseEndpoint;
+            });
+        });
+    }
+
+    internal static IOpenTelemetryBuilder UseOtlpExporter(
+        this IOpenTelemetryBuilder builder,
+        Action<OtlpExporterBuilder> configure)
+    {
+        Guard.ThrowIfNull(configure);
+
+        return UseOtlpExporter(builder, name: null, configure: configure);
+    }
+
+    internal static IOpenTelemetryBuilder UseOtlpExporter(
+        this IOpenTelemetryBuilder builder,
+        IConfiguration configuration)
+    {
+        Guard.ThrowIfNull(configuration);
+
+        return UseOtlpExporter(builder, name: null, configuration: configuration);
+    }
+
+    internal static IOpenTelemetryBuilder UseOtlpExporter(
+        this IOpenTelemetryBuilder builder,
+        string? name = null,
+        IConfiguration? configuration = null,
+        Action<OtlpExporterBuilder>? configure = null)
+    {
+        Guard.ThrowIfNull(builder);
+
+        if (configuration != null && string.IsNullOrEmpty(name))
+        {
+            name = "otlp";
+        }
+
+        var otlpBuilder = new OtlpExporterBuilder(builder.Services, name, configuration);
+
+        configure?.Invoke(otlpBuilder);
+
+        UseOtlpExporterInternal(builder, name);
+
+        return builder;
+    }
+
+    private static void UseOtlpExporterInternal(IOpenTelemetryBuilder builder, string? name)
+    {
+        // Note: We automatically turn on signals for "UseOtlpExporter"
+        builder
+            .WithLogging()
+            .WithMetrics()
+            .WithTracing();
+
+        var services = builder.Services;
+
+        // Note: UseOtlpExporterRegistration is added to the service collection
+        // to detect repeated calls to "UseOtlpExporter" and to throw if
+        // "AddOtlpExporter" extensions are called
+        services.AddSingleton<UseOtlpExporterRegistration>();
+
+        services.RegisterOptionsFactory(configuration => new SdkLimitOptions(configuration));
+        services.RegisterOptionsFactory(configuration => new ExperimentalOptions(configuration));
+        services.RegisterOptionsFactory((sp, configuration, name) => new OtlpExporterBuilderOptions(
+            configuration,
+            sp.GetRequiredService<IOptionsMonitor<SdkLimitOptions>>().CurrentValue,
+            sp.GetRequiredService<IOptionsMonitor<ExperimentalOptions>>().CurrentValue,
+            sp.GetService<IOptionsMonitor<LogRecordExportProcessorOptions>>()?.Get(name),
+            sp.GetService<IOptionsMonitor<MetricReaderOptions>>()?.Get(name),
+            sp.GetService<IOptionsMonitor<ActivityExportProcessorOptions>>()?.Get(name)));
+
+        name ??= Options.DefaultName;
```

Comment:
For my understanding, why do we not need the `??=` assignment for the `sp.GetService<T>()?.Get(name)` calls above?

Reply:
That is a mistake! I moved it up: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5400/commits/6f2c8e012d119b7988e3ffe8b43924ce1c713b39

The options code seems smart enough to treat `null` name as `Options.DefaultName` (ex: https://github.com/dotnet/runtime/blob/cbca5083d3e69f2bd25e397f8894d94d7763a13a/src/libraries/Microsoft.Extensions.Options/src/OptionsCache.cs#L35) so we might actually be able to remove it completely. But I'm going to leave in there for now I think better safe than sorry. We do that a lot of spots so maybe a cleanup PR to remove it everywhere at some point, need to verify all the versions of Options API do it correct which is kind of heavy lifting 🤣